python/ray/train/torch.py

```diff
@@ -297,55 +405,22 @@ def prepare_data_loader(
         move_to_device (bool): If set, automatically move the data
             returned by the data loader to the correct device.
     """
+    return get_accelerator(TorchAccelerator).prepare_data_loader(
+        data_loader,
+        add_dist_sampler=add_dist_sampler,
+        move_to_device=move_to_device,
+    )
 
-    # Only add Distributed Sampler if the following conditions hold:
-    # 1. More than one training worker is being used.
-    # 2. A DistributedSampler has not already been added by the user.
-    # 3. The dataset is not an IterableDataset. Samplers do not worker with
-    # IterableDatasets.
-    if (
-        train.world_size() > 1
-        and not isinstance(data_loader.sampler, DistributedSampler)
-        and not (
-            hasattr(data_loader, "dataset")
-            and isinstance(data_loader.dataset, IterableDataset)
+
+@PublicAPI(stability="beta")
+def accelerate() -> None:
+    """Enables training optimizations."""
+    try:
+        set_accelerator(TorchAccelerator())
+    except RuntimeError:
+        raise RuntimeError(
+            "An accelerator has already been set. Try moving the `accelerate` call to the top of your training function."
```

Comment:
Should we also specify the following?

`Make sure train.torch.accelerate() is not called multiple times, and is called before any of the prepare methods.`